xml_struct_derive/src/properties.rs

```diff
@@ -0,0 +1,328 @@
+        if has_namespace_decl && properties.should_serialize_as_text {
+            // There's no meaningful way to namespace text content, so the
+            // combination of these properties is almost certainly a mistake.
+            errors.push(Error::new(
+                attr.span(),
+                "cannot declare namespaces for text content",
+            ));
+        }
+
+        if properties.ns_prefix_for_variants.is_some() && properties.should_serialize_as_text {
+            errors.push(Error::new(
+                attr.span(),
+                "cannot declare variant namespace prefix for text enum",
+            ));
+        }
+
+        // Combine and return errors if there are any. If none, we've
+        // successfully parsed the attributes and can return the appropriate
+        // props.
+        match errors.into_iter().reduce(|mut combined, err| {
+            combined.combine(err);
+
+            combined
+        }) {
+            Some(err) => Err(err),
+            None => Ok(properties),
+        }
+    }
+}
+
+#[derive(Debug)]
+/// A declaration of an XML namespace for a type with a derived `XmlSerialize`
+/// implementation.
+pub(crate) struct NamespaceDecl {
+    pub prefix: TokenStream,
+    pub name: TokenStream,
+}
+
+#[derive(Debug, Default)]
+/// Properties governing the serialization of a field in a struct or enum with a
+/// derived `XmlSerialize` implementation.
+pub(crate) struct FieldProps {
+    /// The type of XML structure which the field represents.
+    pub repr: FieldRepr,
+
+    /// Whether the field should be serialized with a "flat" representation.
+    ///
+    /// A flattened field will be serialized only as its content nodes, rather
+    /// than as an XML element containing those content nodes.
+    pub should_flatten: bool,
+
+    /// A prefix to add to this field's name when serialized as an element or
+    /// attribute.
+    pub namespace_prefix: Option<TokenStream>,
+}
+
+impl FieldProps {
+    /// Constructs a set of serialization properties for an enum or struct field
+    /// from its struct attributes.
+    pub(crate) fn try_from_attrs(
+        value: Vec<Attribute>,
+        is_named_field: bool,
+    ) -> Result<Self, Error> {
+        // Find the attribute for configuring behavior of the derivation, if
+        // any.
+        let attr = match value
+            .into_iter()
+            .find(|attr| attr.path().is_ident(MACRO_ATTRIBUTE))
+        {
+            Some(attr) => attr,
+
+            // If we don't find a matching attribute, we assume the default set
+            // of properties.
+            None => return Ok(Self::default()),
+        };
+
+        // We build a list of errors so that we can combine them later and emit
+        // them all instead of only emitting the first.
+        let mut errors = Vec::new();
+
+        // We start with the default set of properties, then parse the
+        // `xml_struct` attribute to change any properties which deviate from
```

Comment:
nitpick: Not a typo but using the weak form of "any" in this sentence sounds weird to me.

```suggestion
        // `xml_struct` attribute to change any property which deviate from
```

Reply:
Huh. Never heard this referred to as a weak form. Either one sounds fine to my ear, so I've gone ahead and incorporated your suggestion where appropriate.

Reply:
I think https://dictionary.cambridge.org/us/grammar/british-grammar/any is where I got the "weak"/"strong" form think from

Reply:
Interesting, thanks!